src/vs/editor/common/controller/textAreaState.ts

```diff
@@ -163,7 +163,7 @@ export abstract class TextAreaState {
 		if (currentSelectionStart === currentSelectionEnd) {
 			// composition accept case
 			// [blahblah] => blahblah|
-			if (previousValue === currentValue && previousSelectionStart === 0 && previousSelectionEnd === previousValue.length && currentSelectionStart === currentValue.length) {
+			if (previousValue === currentValue && previousSelectionStart === 0 && previousSelectionEnd === previousValue.length && currentSelectionStart === currentValue.length && currentValue !== '\n') {
```

Comment:
@alexandrudima 

When replacing the end-of-line with a newline, this check results in no typing being recognized (well, on the first typing, on the second it is recorded).  I added this specific check for enter but I'm not convinced it's correct.